tensorboard/plugins/profile/profile_plugin.py

```diff
@@ -327,12 +329,32 @@ def data_route(self, request):
       return http_util.Respond(request, '404 Not Found', 'text/plain', code=404)
     return http_util.Respond(request, data, 'application/json')
 
+  @wrappers.Request.application
+  def capture_route(self, request):
+    service_addr = request.args.get('service_addr')
+    duration = int(request.args.get('duration'))
+    is_tpu_name = request.args.get('is_tpu_name') == 'true'
+    if is_tpu_name:
+      tpu_cluster_resolver = (
+        tf.contrib.cluster_resolver.TPUClusterResolver([service_addr]))
+      service_addr = tpu_cluster_resolver.get_master()
+    service_addr = service_addr.replace('grpc://', '').replace(':8470', ':8466')
+    try:
+      profiler_client.start_tracing(service_addr, self.logdir, duration)
+      return http_util.Respond(
+          request, {'result': 'Capture profile successfully. Please refresh'},
+          'application/json')
+    except tf.errors.UnavailableError:
+      return http_util.Respond(request, {'result': 'Failed to capture profile'},
```

Comment:
Hmm please use proper response status code for such error. 404 would be appropriate AFAICT